core/src/main/scala/org/bitcoins/core/wallet/utxo/AddressTag.scala

```diff
@@ -0,0 +1,44 @@
+package org.bitcoins.core.wallet.utxo
+
+/** A type of address tag, many AddressTags of the same type
+  * should inherit the AddressTagType that they all share
+  */
+trait AddressTagType {
+  def name: String
+}
+
+/**
+  * An tag for an address. It's name is what it is referred to as
+  * and it's tagType is its parent AddressTagType
+  */
+trait AddressTag {
+  def name: String
+  def tagType: AddressTagType
+
+  def ==(at: AddressTag): Boolean =
+    name == at.name && tagType.name == at.tagType.name
+  def !=(at: AddressTag): Boolean = !(this == at)
+
+  def toUnknown: UnknownAddressTag = UnknownAddressTag(name, tagType)
```

Comment:
shouldn't this belong in `ExternalAddressTag` rather than in the top-level type? I can't imagine when it would be right to convert a known internal type to unknown

Reply:
It could but I don't see the harm of it existing here